### PR TITLE
CMR-4134 - Added UMM-Var schema and models.

### PR DESCRIPTION
**Pull request by oubiwann**
This is part 1 of a three-parter for UMM-Var ingest:

1. Add schema and models for variables (with some placeholders)
2. Add validation code for variables (with some placeholders)
3. Add ingest-services and routes for variables

Future changes in the evolving UMM-Var spec won't have too big an impact on this code, since there's not a log here. It will involve updating the JSON file, regenerating the model, and maybe tweaking the minimal test that's in place. 

Once the work on the spec is ready for release, we'll come back to this code and fill more of it in. In the mean time, it gets us to a place where we can at least use it in other parts of the code (e.g., validation).

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj [hunk 1]
```diff
@@ -62,3 +63,24 @@
...
+
+(defn validate-variable
+  "Validates the UMM record returning a list of error maps containing a path
+  through the UMM model and a list of errors at that path. Returns an empty
+  sequence if it is valid."
+  ([variable]
+   (validate-variable variable nil))
+  ([variable additional-validations]
+   (validation-errors->path-errors
+     (v/validate (cons vv/variable-validations additional-validations)
```
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj]: Whoops -- two spaces there; I'll fix that.

File: umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj [hunk 2]
```diff
@@ -62,3 +63,24 @@
...
+                 variable))))
+
+(defn validate-variable-warnings
+  "Validates the UMM record against the list of warnings - issues that we want
+  to convey to the user, but which we do not consider failures."
+  ([variable]
+   (validate-variable-warnings variable nil))
+  ([variable additional-validations]
+   (validation-errors->path-errors
+     (v/validate (cons vv/variable-validation-warnings additional-validations)
```
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj]: Another two-space indent; will fix.

File: umm-lib/src/cmr/umm/umm_variable.clj [hunk 3]
```diff
@@ -0,0 +1,21 @@
+(ns cmr.umm.umm-variable
+  "Defines the UMM Variable record."
+  (:require
+   [cmr.common.dev.record-pretty-printer :as record-pretty-printer]))
+
+(defrecord UmmVariable
```
Comment (indiejames) [inline umm-lib/src/cmr/umm/umm_variable.clj]: Is this necessary? I'm not sure umm-lib records are needed any more.

Comment (indiejames) [inline umm-lib/src/cmr/umm/umm_variable.clj]: At least not new ones.

File: umm-lib/src/cmr/umm/umm_core.clj [hunk 4]
```diff
@@ -1,24 +1,28 @@
...
-            [cmr.umm.iso-mends.iso-mends-collection :as iso-mends-c]
-            [cmr.umm.iso-mends.granule :as iso-mends-g]
-            [cmr.umm.iso-smap.iso-smap-core :as iso-smap]
-            [cmr.umm.iso-smap.iso-smap-collection :as iso-smap-c]
-            [cmr.umm.iso-smap.granule :as iso-smap-g]
-            [cmr.umm.umm-collection]
-            [cmr.umm.umm-granule])
-  (:import cmr.umm.umm_collection.UmmCollection
-           cmr.umm.umm_granule.UmmGranule))
+  (:require
```
Comment (ygliuvt) [inline umm-lib/src/cmr/umm/umm_core.clj]: We don't need to change `umm-lib` for new concept types. New concept types should only be added in `umm-spec-lib`.

File: umm-lib/src/cmr/umm/umm_variable.clj (hunk 3 above)
Comment (oubiwann) [inline umm-lib/src/cmr/umm/umm_variable.clj]: Good to know -- thanks!

File: umm-lib/src/cmr/umm/umm_core.clj (hunk 4 above)
Comment (oubiwann) [inline umm-lib/src/cmr/umm/umm_core.clj]: Thanks!

File: umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj (hunk 1 above)
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj]: Fixed.

File: umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj (hunk 2 above)
Comment (oubiwann) [inline umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_validation_core.clj]: Fixed.

File: umm-lib/src/cmr/umm/umm_variable.clj (hunk 3 above)
Comment (oubiwann) [inline umm-lib/src/cmr/umm/umm_variable.clj]: Removed.

File: umm-spec-lib/test/cmr/umm_spec/test/validation/umm_spec_variable_validation_tests.clj [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  (:require
+   [clojure.test :refer :all]
+   [cmr.common.util :as u :refer [are3]]
+   [cmr.umm-spec.models.umm-common-models :as cm]
+   [cmr.umm-spec.models.umm-variable-models :as vm]
+   [cmr.umm-spec.test.validation.umm-spec-validation-test-helpers :as helpers]
+   [cmr.umm-spec.validation.umm-spec-validation-core :as v]
+   [cmr.umm.umm-variable :as variable]))
+
+(deftest umm-variable-creation
```
Comment (oubiwann) [inline umm-spec-lib/test/cmr/umm_spec/test/validation/umm_spec_variable_validation_tests.clj]: Need to switch this out to use the generated record ...

Comment (oubiwann) [inline umm-spec-lib/test/cmr/umm_spec/test/validation/umm_spec_variable_validation_tests.clj]: Done. Re-running tests ...

**Comment by oubiwann** [comment]
Thanks, all!

**Pull request merged**